packages/clients/src/internal/interceptors/types.ts

```diff
@@ -0,0 +1,17 @@
+export interface RequestInterceptor {
+  (request: Readonly<Request>): Request | Promise<Request>
```

Comment:
it's possible to prefer one object instead of params ? 
```suggestion
  ({ request }: { request : Readonly<Request> }): Request | Promise<Request>
```